### ISSUE TITLE
API bug? A user on Windows (20H2) Firefox is getting a static mono font when they request a variable MONO axis

### ISSUE THREAD

**Issue by arrowtype** [google/fonts#3321]
Bringing this over from https://github.com/arrowtype/recursive/issues/460.

## Problem

@subnut is using a valid GF API URL to request Recursive with the `MONO` axis:

`https://fonts.googleapis.com/css2?family=Recursive:MONO@0..1&display=swap`

The URL they are using works fine for me in a test on CodePen, but the same CodePen only returns a static mono font for them. A screenshot from them:

![image](https://user-images.githubusercontent.com/45946693/115084313-b44bf980-9ed6-11eb-9a7c-64c51ed74d36.png)

If I download fonts from the font URLs in the CSS @subnut gets vs from the CSS I get, I do see that they are getting Monospace, static fonts, while I am getting fonts with a variable MONO axis.

![image](https://user-images.githubusercontent.com/45946693/115084424-e3fb0180-9ed6-11eb-9dd4-793b3b0cf2cb.png)

## Version info

The fonts returned are Recursive v `1.047`. So, this is a problem from before this [week’s merged update of Recursive](https://github.com/google/fonts/pull/3279) (which probably hasn’t made it onto the GF servers quite yet).

## My best guess at what is going wrong

My best guess is, the Google Fonts API might be guessing wrong about what CSS to send @subnut’s OS/browser, and is probably sending CSS that would be sent to a system that couldn’t handle variable fonts. But, they are on a very recent version of Windows & Firefox, so that is surprising. Maybe Firefox is somehow masking their info, for privacy reasons?

**Comment by subnut** [comment]
Side note:

@arrowtype 
I just checked on Google Chrome on the same PC, and it renders correctly on Google Chrome. The CSS I received matches the one you showed earlier.

**Comment by thlinard** [comment]
Very interesting. More generally, I think errors in detecting which browsers can or cannot handle VFs is one of the reasons for the persistence of  issue #2602 (there are other reasons, because I believe that people report seemingly similar issues that actually have different causes, but that's another problem).

**Comment by arrowtype** [comment]
@thlinard I’m fairly sure that is a totally separate issue. Mac messes up black/white rendering of some glyphs (like A) that have two contours and overlap. https://twitter.com/ArrowType/status/1311716316320018438

This issue is that the GF API seems to be serving the wrong CSS (and therefore the wrong fonts) to Windows Firefox.

**Comment by thlinard** [comment]
@arrowtype Well, #2602 is neither an issue related to the latest versions of macOS (perfectly capable of handling VFs, including with overlap), nor to the latest versions of Firefox (ditto).


In fact, "GF API seems to be serving the wrong CSS (and therefore the wrong fonts)" describes well enough #2602 (fonts intended for very old browsers are sent to newer browsers).

**Comment by subnut** [comment]
I am curious, how exactly does Google Fonts detect if the browsers supports Variable fonts? Does it decide solely based on the User Agent, or is there anything else involved?

**Comment by arrowtype** [comment]
Good question! I’m not sure whether they have/would publish that info.

**Comment by arrowtype** [comment]
Hmm, I think this might be related to issue https://github.com/google/fonts/issues/2211.

It seems that Google Fonts might be interpreting Firefox as not being able to handle variable fonts, and therefore sending statics. This probably interacts with the API badly for some of the newer axes, like `MONO`. 

This is a total guess, but I suppose this might be due to Firefox spoofing user agent information, e.g. to avoid ad tracking?

**Comment by toothbrush7777777** [comment]
> 
> 
> It seems that Google Fonts might be interpreting Firefox as not being able to handle variable fonts, and therefore sending statics. This probably interacts with the API badly for some of the newer axes, like `MONO`.
> 
> This is a total guess, but I suppose this might be due to Firefox spoofing user agent information, e.g. to avoid ad tracking?

No, Google Fonts does not serve variable fonts when the user agent is Firefox. See issue #2543 for their original (outdated) reasoning.
